src/internal/minikubetestenv/deploy.go

```diff
@@ -878,13 +880,50 @@ func putRelease(t testing.TB, ctx context.Context, namespace string, kubeClient
 	previousOptsHash := getOptsConfigMapData(t, ctx, kubeClient, namespace)
 	pachdExists, err := checkForPachd(t, ctx, kubeClient, namespace, opts.EnterpriseServer)
 	require.NoError(t, err)
-	if mustUpgrade {
+	if mustUpgrade || opts.InstallPrometheus {
 		t.Logf("Test must upgrade cluster in place, upgrading cluster with new opts in %v", namespace)
 		require.NoErrorWithinTRetry(t,
 			time.Minute,
 			func() error {
 				return errors.EnsureStack(helm.UpgradeE(t, helmOpts, chartPath, namespace))
 			})
+		// get latest version with: helm pull prometheus-community/kube-prometheus-stack -d etc/helm/charts/
```

Comment:
I don't think minikubetestenv needs to know about installing Prometheus.  Can your test just call helm.UpgradeE after AcquireCluster returns?  (src/server/proxy_test does something similar.)

Reply:
It could, but I think it would be odd for minikubetestenv to allocate a port for something it doesn’t manage.